test/powershell/Modules/Microsoft.PowerShell.Utility/WebCmdlets.Tests.ps1

```diff
@@ -715,7 +715,8 @@ Describe "Invoke-WebRequest tests" -Tags "Feature" {
 
     It "Validate Invoke-WebRequest body is converted to query params for CustomMethod GET" {
 
-        $command = "Invoke-WebRequest -Uri 'http://httpbin.org/get' -CustomMethod GET -Body @{'testparam'='testvalue'}"
+        $uri = Get-WebListenerUrl -Test 'Get'
+        $command = "Invoke-WebRequest -Uri '$uri' -CustomMethod GET -Body @{'testparam'='testvalue'}"
```

Comment:
Typo  '$uri' ->  $uri

Reply:
I am not a big fan of unencapsulated strings in evaluated commands. the command would become

```powershell
Invoke-WebRequest -Uri http://localhost:8083/Get -CustomMethod GET -Body @{'testparam'='testvalue'}
```

Which is improper IMO. Does this have to be done? I feel like this change was actually an improvement and I made it everywhere purposefully.

Reply:
Binding is not preprocessor, we don't parse the variable value, so we shouldn't quote variables. `Uri` get right string.
My thoughts was that we should use the file pattern.

Reply:
In this instance the `$Command` string is later evaluated by `Invoke-Expression`. `$uri` is expanded before that evaluation. In instances where the url may contain a `$`, it would be expanded first during the `$command` assignment and then when the `Invoke-Expression` is run on `$command` the `$`  in the url would attempt to be expanded again.

```powershell
# $Field1 is a literal field name, not a variable
$uri = 'http://localhost:8083/Get?$Field1=Value1'
$Command = "Write-Output $uri"
Invoke-Expression $Command
```

Result:
```
http://localhost:8083/Get?=Value1
```

vs:

```powershell
$uri = 'http://localhost:8083/Get?$Field1=Value1'
$Command = "Write-Output '$uri'"
Invoke-Expression $Command
```

Result:

```
http://localhost:8083/Get?$Field1=Value1
```

Reply:
Sorry again - I lost the context.
Closed.